engine/apps/api/serializers/organization.py

```diff
@@ -109,7 +110,26 @@ def get_banner(self, obj):
 
     def get_limits(self, obj):
         user = self.context["request"].user
-        return obj.notifications_limit_web_report(user)
+        if not settings.OSS_INSTALLATION:
+            return obj.notifications_limit_web_report(user)
+
+        # show a version warning on OSS installations in case backend and frontend are different versions
+        frontend_version = self.context["request"].headers.get("X-OnCall-Plugin-Version")
+        backend_version = settings.VERSION
+        version_warning = {}
+        if backend_version and frontend_version and backend_version != frontend_version:
+            text = (
+                "Version mismatch! Please make sure you have the same versions of Grafana OnCall plugin "
+                "and Grafana OnCall engine, "
+                "otherwise there could be issues with your Grafana OnCall installation! "
+                f"Current plugin version: {frontend_version}, current engine version: {backend_version}. "
+                "To install the latest Grafana OnCall plugin version, "
+                "visit the plugin configuration page and click Update. "
+                "To update Grafana OnCall engine, re-pull the grafana/oncall image and restart (or see the docs)."
+            )
+            version_warning = {"period_title": "Version mismatch", "show_limits_warning": True, "warning_text": text}
```

Comment:
@vadimkerr are you sure it should be a part of `get_limits` method?

Reply:
It shouldn't, but the idea is that this warning should pop up on new backend version and old frontend version. So we need to show a warning without changing the frontend, it's the only place where we can show a warning generated on backend I think. Are there any other places on backend that I forgot about?

Reply:
I think for now it's ok, but we need to refactor it later.